src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -307,25 +309,37 @@ static void OnGetAddrInfo(DNSServiceRef sdRef, DNSServiceFlags flags, uint32_t i
     {
         VerifyOrReturn(sdCtx->HasAddress(), sdCtx->Finalize(kDNSServiceErr_BadState));
 
-        if (domainName.compare(kOpenThreadDot) == 0)
+        if (domainName.compare(kSrpDot) == 0)
         {
-            ChipLogProgress(Discovery, "Mdns: Resolve completed on the open thread domain.");
+            ChipLogProgress(Discovery, "Mdns: Resolve completed on the srp domain.");
+
+            // Cancel the timer if one has been started
+            if (sdCtx->hasSrpTimerStarted)
+            {
+                CancelSrpTimer(sdCtx);
+            }
             sdCtx->Finalize();
         }
         else if (domainName.compare(kLocalDot) == 0)
         {
-            ChipLogProgress(
-                Discovery,
-                "Mdns: Resolve completed on the local domain. Starting a timer for the open thread resolve to come back");
+            ChipLogProgress(Discovery,
+                            "Mdns: Resolve completed on the local domain. Starting a timer for the srp resolve to come back");
 
-            // Usually the resolution on the local domain is quicker than on the open thread domain. We would like to give the
-            // resolution on the open thread domain around 250 millisecs more to give it a chance to resolve before finalizing
+            // Usually the resolution on the local domain is quicker than on the srp domain. We would like to give the
+            // resolution on the srp domain around 250 millisecs more to give it a chance to resolve before finalizing
             // the resolution.
-            if (!sdCtx->hasOpenThreadTimerStarted)
+            if (!sdCtx->hasSrpTimerStarted)
             {
-                // Schedule a timer to allow the resolve on OpenThread domain to complete.
-                StartOpenThreadTimer(kOpenThreadTimeoutInMsec, sdCtx);
-                sdCtx->hasOpenThreadTimerStarted = true;
+                // Schedule a timer to allow the resolve on Srp domain to complete.
+                CHIP_ERROR error = StartSrpTimer(kSrpTimeoutInMsec, sdCtx);
+
+                // If the timer fails to start, finalize the context and return.
```

Comment:
This comment should explain _why_, not _what_.  The "what" (finalize and return) is obvious from the code below.  The question is: why is that the right thing to do?  (It is, but the comment should explain why.)

Reply:
will do.